### DISCUSSION TITLE
new variable ui not very friendly for long metric queries

### DISCUSSION THREAD

**Discussion by nbuesing** [grafana/grafana#64657] [Dashboards, Panels & Visualizations]
`kafka_streams_stream_task_metrics_process_rate{cluster_id=~"$cluster", instance=~"${instance}",thread_id=~"${thread}"}
`

there is a lot of screen real estate for variables, it would be nice if the above query could be shown. 
Also, would like to be able to stay in "advanced" mode and have the old functionality back.  Is there a way to do that?

I would rather be able to still create my variable writing out the query in full, like this, and avoid the UI all together.

`label_values(kafka_streams_stream_task_metrics_process_rate{cluster_id=~\"$cluster\", instance=~"$instance",thread_id=~"$thread"}, task_id)`

I keep enjoying the improvements to Grafana, just wanting to share that the new UI doesn't seem to scale well when you have variables built on other variables and want to limit them through a "drill-down" selection behavior; thanks.

<img width="1162" alt="image" src="https://user-images.githubusercontent.com/1424057/224551447-7d190b2b-d871-4963-86c9-ec4d7a8f7bf5.png">

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82848. Feel free to continue the discussion around this there. Thank you!
